node/src/store/redb_store.rs

```diff
@@ -316,21 +321,28 @@ impl RedbStore {
     }
 
     async fn get_sampling_metadata(&self, height: u64) -> Result<Option<SamplingMetadata>> {
+        if !self.contains_height(height).await {
+            return Err(StoreError::NotFound);
+        }
+
```

Comment:
This opens a read TX and later we open another one. Both should be done in the same read TX.

Reply:
Thought I could get away with it, since we're add-only :P, but you're right doing it in one txn would be better. Fixed 👍